server/monitor/src/main/java/org/apache/accumulo/monitor/servlets/trace/ShowTrace.java

```diff
@@ -45,9 +46,11 @@
   private static final long serialVersionUID = 1L;
   private static final String checkboxIdSuffix = "_checkbox";
   private static final String pageLoadFunctionName = "pageload";
+  private static final Pattern TRACE_ID_PATTERN = Pattern.compile("\\p{XDigit}{16}");
 
   String getTraceId(HttpServletRequest req) {
-    return getStringParameter(req, "id", null);
+    final String stringValue = getStringParameter(req, "id", null);
+    return TRACE_ID_PATTERN.matcher(stringValue).matches() ? stringValue : null;
```

Comment:
This looks like it could produce an NPE from the matcher if `stringValue` is null. We probably want to add a check for that:

```suggestion
    if (stringValue == null) {
      return null;
    }
    return TRACE_ID_PATTERN.matcher(stringValue).matches() ? stringValue : null;
```